packages/lib/src/components/Dropin/components/DropinComponent.tsx

```diff
@@ -92,6 +92,25 @@ export class DropinComponent extends Component<DropinComponentProps, DropinCompo
         this.setState({ activePaymentMethod: null });
     }
 
+    /**
+     * onOrderCancelBuilder decides which onOrderCancel logic should be used, manual or sessions
+     */
+    private onOrderCancelBuilder = () => {
```

Comment:
I think it is a bit overcomplicating to create a builder to return the function that will cancel the gitftcard order. It is also not very efficient as every render you inject a new function into the `<PaymentMethodList />`  when `this.onOrderCancelBuilder()` is called, therefore forcing the render.

Why not create normal handler function to do that and pass it as callback? Is there any problem doing that?

Ex:
```ts
private handleOnOrderCancel = async (data: OrderData): Promise<void> => {
  if (this.props.session) {
    try {
       await this.props.session.cancelOrder(data);
       this.props._parentInstance.update({order: null});
      } catch (error) {
        this.setStatus(error?.message || 'error');
      }
      return;
    }    
    if (this.props.onOrderCancel) this.props.onOrderCancel(data);
}
```

Reply:
I agree with the performance concern. I can move the function call to the constructor and that would make more sense. 
That being said, the logic in that function is different, because for several reasons:
1. We are always passing an function to the component PaymentMethodList and that makes the button will always be active even when there is not a valid implementation of the function (ie: there is no `onOrderCancel` and not in sessions)
2. Sessions in that case take precedence over onOrderCancel, which I don't think makes sense since the merchant can still chose to implement his own logic for onOrderCancel

Reply:
1. A button being active just because it has a callback sounds a bit off. Might be worth to revisit this logic and see if it can be improved
2. Can merchant that is using sessions implement his own `onOrderCancel`? I am not sure if that makes sense from a merchant perspective. From my understanding the session endpoint is called internally by the SDK only. I don't see the use-case for a merchant providing its own `onOrderCancel`. Let me know what I am missing